app/src/main/java/com/google/android/sambadocumentsprovider/nativefacade/SambaMessageLooper.java

```diff
@@ -55,7 +58,12 @@ private void init() {
       mLatch.await();
 
       mCredentialCacheClient = new CredentialCacheClient(mLooper, mCredentialCacheImpl);
-      mServiceClient = new SambaFacadeClient(mLooper, mClientImpl);
+
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
It's the right choice to have different implementation for different versions of Android, but I would like to do it on SmbFile -- because its implementation is still roughly the same.